### PR TITLE
sdk(react-native-sdk): error fixes

### PR DESCRIPTION
**Pull request by Calinteodor**

### CODE REVIEW

File: react/features/mobile/background/middleware.native.ts [hunk 1]
```diff
@@ -74,8 +74,11 @@ function _setAppStateListenerF({ getState }: IStore, next: Function, action: Any
     const result = next(action);
     const { appStateListener: newListener } = getState()['features/background'];
 
+    // @ts-ignore
+    const appStateOldListener = AppState.addEventListener('change', oldListener);
```
Comment (saghul) [inline react/features/mobile/background/middleware.native.ts:78]: This won't remove the previous listener. We need to store the return value of AppState.addEventListener and place that in redux.

**Review by saghul — CHANGES_REQUESTED** [review]

File: react/features/mobile/background/middleware.native.ts (hunk 1 above)
Comment (Calinteodor) [reply react/features/mobile/background/middleware.native.ts]: I created an action(eventSubcription) that stores the previous appStateListener.

**Comment by saghul** [comment]
You can simplify things by storing only the subscription and not the listener.

**Comment by saghul** [comment]
Can you pl rename things along the lines of _SET_APP_STATE_SUBSCRIPTION then?

_ means it's a private action
Action names being with a verb, so SET.
And last, the name is more descriptive.

**Review by saghul — APPROVED** [review]

**Pull request merged**